projects/ZenoFX/ne.cpp

```diff
@@ -0,0 +1,149 @@
+//
+// Created by admin on 2022/6/17.
+//
+#include <zeno/zeno.h>
+#include <zeno/types/StringObject.h>
+#include <zeno/types/NumericObject.h>
+#include <zfx/zfx.h>
+#include <zfx/x64.h>
+#include <cassert>
+#include <vector>
+#include "dbg_printf.h"
+
+namespace {
+static zfx::Compiler compiler;
+static zfx::x64::Assembler assembler;
+
+static void numeric_eval (zfx::x64::Executable *exec,
+                         std::vector<float> &chs) {
+    auto ctx = exec->make_context();
+    for (int j = 0; j < chs.size(); j++) {
+        ctx.channel(j)[0] = chs[j];
+    }
+    ctx.execute();
+    for (int j = 0; j < chs.size(); j++) {
+        chs[j] = ctx.channel(j)[0];
+    }
+
+}
+/*
+ * 接收一个Numericfloat的参数，makedict一下传给$
+ * */
+struct NumericEval : zeno::INode {
+    virtual void apply() override {
+        auto code = get_input<std::string>("zfxCode")->get();
+        //auto code = get_params<>
+        //一个模板函数，返回一个std::shared_ptr<T>，这里对这一个智能指针调用get()返回一个裸指针
+       // auto op = get_param<std::string>("op_type");
+       // auto FrameId = std::make_shared<zeno::NumericObject>();
+        //auto FrameTime = std::make_shared<zeno::NumericObject>();
+
+
+        zfx::Options opts(zfx::Options::for_x64);
+        opts.detect_new_symbols = true;
+//现在有一个问题就是NumericEval如果只接收一个std::string，那么用户输入zfx代码中包含$frame，我们如何设置这一个$DictObject的值
+        auto params = std::make_shared<zeno::DictObject>();
+        //params->
```

Comment:
ruiping: add 'frame' here.
```suggestion
        params->lut["frame"] = getGlobalState()->frameid;
```